packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -11221,95 +11221,115 @@ void main() {
+        await tester.pumpAndSettle();
+        maxScrollExtent = scrollController.position.maxScrollExtent;
+        expect(scrollController.offset, targetOffset);
+      }
 
-    await tester.pumpWidget(
-      MaterialApp(
-        home: Scaffold(
-          body: Center(
-            child: SizedBox(
-              height: 32,
-              child: EditableText(
-                key: key,
-                focusNode: focusNode,
-                style: Typography.material2018().black.subtitle1!,
-                cursorColor: Colors.blue,
-                backgroundCursorColor: Colors.grey,
-                controller: controller,
-                scrollController: scrollController,
-                maxLines: 2,
+
+      Future<TextSelectionDelegate> pumpLongScrollableText(WidgetTester tester) async {
+        final GlobalKey<EditableTextState> key = GlobalKey<EditableTextState>();
+
+        await tester.pumpWidget(
+          MaterialApp(
+            home: Scaffold(
+              body: Center(
+                child: SizedBox(
+                  height: 32,
+                  child: EditableText(
+                    key: key,
+                    focusNode: focusNode,
+                    style: Typography.material2018().black.subtitle1!,
+                    cursorColor: Colors.blue,
+                    backgroundCursorColor: Colors.grey,
+                    controller: controller,
+                    scrollController: scrollController,
+                    maxLines: 2,
+                  ),
+                ),
               ),
             ),
           ),
-        ),
-      ),
-    );
+        );
 
-    final TextSelectionDelegate textSelectionDelegate = key.currentState!;
+        /// Populate [maxScrollExtent]
+        await resetSelectionAndScrollOffset(tester);
+        return key.currentState!;
+      }
 
-    late double maxScrollExtent;
-    Future<void> resetSelectionAndScrollOffset([bool setMaxScrollExtent = true]) async {
-      controller.value = controller.value.copyWith(
-        text: text,
-        selection: controller.selection.copyWith(baseOffset: 0, extentOffset: 1),
-      );
+    testWidgets('SelectAll toolbar action will not set max scroll on iOS', (WidgetTester tester) async {
+      final TextSelectionDelegate textSelectionDelegate = await pumpLongScrollableText(tester);
+
+      await resetSelectionAndScrollOffset(tester, false);
+      textSelectionDelegate.selectAll(SelectionChangedCause.toolbar);
       await tester.pump();
-      final double targetOffset = setMaxScrollExtent ? scrollController.position.maxScrollExtent : 0.0;
-      scrollController.jumpTo(targetOffset);
-      await tester.pumpAndSettle();
-      maxScrollExtent = scrollController.position.maxScrollExtent;
-      expect(scrollController.offset, targetOffset);
-    }
+      expect(scrollController.offset, 0.0);
+    }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS }));
```

Comment:
Nit: Would it be useful to run this test on all platforms and put this final `expect` in a `switch` to show what the different behavior is?  Or maybe that's covered by the next test anyway.